File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.js
const convert = n => {
    if(n === 6) return 1;
    if(n === 5) return 2;
    if(n === 4) return 3;
    if(n === 3) return 4;
    if(n === 2) return 5;
    else return 6;
}

function solution(lottos, win_nums) {
    let unknown = 0, confirmedWin = 0;
    for(let num of lottos) {
        if(num === 0) unknown++;
        else if(win_nums.includes(num)) confirmedWin++;
    }
    const max = unknown + confirmedWin;
    const min = confirmedWin;
    return [convert(max), convert(min)];
}

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    fixed = 0
    zero = 0
    for n in lottos:
        if n in win_nums:
            fixed += 1
            win_nums.remove(n)
        if n == 0:
            zero += 1
            
    def get_rank(n):
        if n == 6:
            return 1
        if n == 5:
            return 2
        if n == 4:
            return 3
        if n == 3:
            return 4
        if n == 2:
            return 5
        else:
            return 6
    
    return [get_rank(fixed + zero), get_rank(fixed)]